src/IncrementalChecker.ts

```diff
@@ -359,4 +363,14 @@ export class IncrementalChecker implements IncrementalCheckerInterface {
       lints.map(this.createNormalizedMessageFromRuleFailure)
     );
   }
+
+  private emitFiles() {
+    if (!this.program) {
+      throw new Error('Invoked called before program initialized');
+    }
+
+    if (this.enableEmitFiles) {
```

Comment:
generally I'd like to use `if (enableEmitFiles)` inside the main body of the checker; it's slightly confusing at present in that `this.emitFiles();` suggests that files are *always* emitted

Reply:
You are right. I started doing this check inside the main body of the checker but a moment later I transferred it inside `this.emitFiles()` to test something and let it there. I have just taken it back to the main body, thank you for your comment John.